src/conditional-compilation.md

```diff
@@ -299,6 +304,25 @@ r[cfg.proc_macro]
 Set when the crate being compiled is being compiled with the `proc_macro`
 [crate type].
 
+r[cfg.version]
+## The `version()` predicate
+
+r[cfg.version.behaviour]
+The `version()` predicate evaluates to true if both:
+
+* The version number contained inside follows the format and
+* The version number contained inside is less than or equal to the version
+  of the language the compiler targets. Usually the compiler version and
+  language version match. So compiler version `1.50.0` targets language
+  `1.50.0`.
+
+r[cfg.version.format]
+In order for it to be considered of valid format, the version number has to
+follow either the `"a.b.c"` scheme or the `"a.b"` scheme, where `a,b,c` are
+decimal integers between `0` and `65535`, inclusively. Semantically, assume `c`
```

Comment:
Question: Does this detail matter in the reference? Isn't it enough to say "these should be integers", and leave the range to the compiler implementation?

Reply:
@traviscross and I discussed this a little bit and felt like this probably *should* be specified? For example, if you say `cfg(version("1.81.65535"))`, that will be true, but `cfg(version("1.81.65536"))` will be false. Since the magnitude of what gets parsed seems to matter, it might be good to be clear about that.

This is a bit extreme, though. Even if an implementation used u8, it's still 19 years before 1.256 (assuming we stick with 6 week releases), and seems silly for someone to put an unnecessarily large number there.

Reply:
hmm yeah good point. For future compatibility we will warn on unknown version literal formats, which includes `1.81.65536`. We should probably word it in a way that the unknown formats are not guaranteed to evaluate to false.